### ISSUE TITLE
ZIO Core: Add whenNotM/ifNotM

### ISSUE THREAD

**Issue by nadenf** [zio/zio#3254]
Given the following code:

```scala
def fileExists = ...

for {
 _ <- ZIO.whenM(fileExists)(createFile)
}
```
Without a whenNotM/ifNotM you will need create a superfluous fileNotExists method.

**Comment by adamgfraser** [comment]
I'm a little worried about the proliferation of methods here. This can be written easily today as:

```scala
for {
  _ <- ZIO.whenM(fileExists.map(!_))(createFile)
}
```

Or:

```scala
for {
  b <- fileExists
  _ <- createFile.when(!b)
{
```

`whenNot` also feels awkward to me. If we do something here maybe we call it `unless`? I'm not sure this is a net positive though given it is one more way to do a very similar thing even though it is slightly more ergonomic in some cases.

**Comment by adamgfraser** [comment]
Closed by #3259.
